### PR TITLE
Add modelarts's training codes

### PR DESCRIPTION
**Pull request by VictorHe-1**
Motivation:
1. Upload the training codes from the ModelArt platform;
2. Combining Luoyang's codes with the latest codes;
3. Add data check codes.

### CODE REVIEW

File: examples/stable_diffusion_v2/tools/data_check/data_check_part.py
Comment (SamitHuang) [inline examples/stable_diffusion_v2/tools/data_check/data_check_part.py:1]: recommend to place in tools/data_utils/.

File: examples/stable_diffusion_v2/ldm/data/dataset.py [hunk 1]
```diff
@@ -63,15 +65,23 @@ def load_data(
         filter_small_size=filter_small_size,
     )
     datalen = dataset.__len__
+
     loader = build_dataloader_ft(dataset, datalen, t2i_collate, batch_size, device_num, rank_id=rank_id)
```
Comment (SamitHuang) [inline examples/stable_diffusion_v2/ldm/data/dataset.py:69]: This part of code is DIFF from Luoyang's implementation as I remembered, which is 
``` python
device_num_per_workers = get_local_rank_size()
shard_id = get_local_rank()
loader = build_dataloader_ft(dataset, datalen, t2i_collate, batch_size, device_num_per_workers , rank_id=shard_id )
...
batchlen = datalen // (batch_size * device_num_per_workers )  ## pay attention here
```

File: examples/stable_diffusion_v2/train_text_to_image.py [hunk 2]
```diff
@@ -52,6 +53,17 @@ def init_env(args):
...
+        print("device_num", device_num, "rank_id", rank_id, "device_num / 8", int(device_num / 8), "rank_id / 8",
+              int(rank_id / 8), flush=True)
+        start_part_idx, start_tar_idx, start_sample_idx, end_part_idx, end_tar_idx, end_sample_idx, tars_to_sync, mapping = calculate_split(
+            args.worker_num, int(rank_id / 8), args.json_data_path)
+        download_tar(tars_to_sync, mapping, start_sample_idx, end_sample_idx, args.json_data_path)
+        print("device_num", device_num, "rank_id", rank_id,
+              "start_part_idx", start_part_idx, "start_tar_idx", start_tar_idx, "start_sample_idx", start_sample_idx,
+              "end_part_idx", end_part_idx, "end_tar_idx", end_tar_idx, "end_sample_idx", end_sample_idx,
+              "tars_to_sync", tars_to_sync, flush=True)
+
```
Comment (SamitHuang) [inline examples/stable_diffusion_v2/train_text_to_image.py:66]: may make it cleaner

File: examples/stable_diffusion_v2/configs/train_config_v2.json
[the remark this answers is not in the document]
Comment (VictorHe-1) [reply examples/stable_diffusion_v2/configs/train_config_v2.json:1]: ok

File: examples/stable_diffusion_v2/configs/train_config_v2.json [hunk 3]
```diff
@@ -7,21 +7,23 @@
...
     "image_size": 512,
     "image_filter_size": 256,
     "random_crop": false,
     "filter_small_size": true,
     "start_learning_rate": 1e-5,
-    "end_learning_rate": 1e-7,
+    "end_learning_rate": 5e-6,
     "decay_steps": 0,
-    "save_checkpoint_steps": 10000
+    "use_ema": true,
```
[the remark this answers is not in the document]
Comment (VictorHe-1) [reply examples/stable_diffusion_v2/configs/train_config_v2.json:26]: Added in train_config_v2.1_base.json

File: examples/stable_diffusion_v2/ldm/data/dataset.py [hunk 4]
```diff
@@ -63,15 +65,23 @@ def load_data(
...
         batchlen = sample_num
     metaloader = MetaLoader(dataloaders, datalen=batchlen, task_num=len(dataloaders.keys()))
-    dataset = GeneratorDataset(metaloader, column_names=data_column, shuffle=True)
 
+    device_num_per_workers = get_local_rank_size()
+    shard_id = get_local_rank()
+    print("device_num_per_workers", device_num_per_workers, "shard_id", shard_id, flush=True)
+    dataset = GeneratorDataset(metaloader, column_names=data_column, shuffle=True,
+                               num_shards=device_num_per_workers,
+                               shard_id=shard_id)
```
[the remark this answers is not in the document]
Comment (VictorHe-1) [reply examples/stable_diffusion_v2/ldm/data/dataset.py]: Changed.

File: examples/stable_diffusion_v2/ldm/data/dataset.py [hunk 5]
```diff
@@ -96,21 +107,39 @@ def list_image_files_captions_recursively(data_path):
...
+    '''
+    csv = pd.read_csv(os.path.join(data_path, "merged_imgp_text.csv"))
+    all_images = list(csv["dir"])
+    all_images = [os.path.join(data_path, f) for f in all_images]
+    all_captions = list(csv["text"])
+    _logger.info(f"Before filter, Total number of training samples: {len(all_images)}")
     return all_images, all_captions
 
 
 def filter_small_image(all_images, all_captions, image_filter_size):
```
[the remark this answers is not in the document]
Comment (VictorHe-1) [reply examples/stable_diffusion_v2/ldm/data/dataset.py]: Add an option `replace` to indicate whether we'll replace or not.

File: examples/stable_diffusion_v2/ldm/data/num_samples_64_part.json
[the remark this answers is not in the document]
Comment (VictorHe-1) [reply examples/stable_diffusion_v2/ldm/data/num_samples_64_part.json:1]: Removed

File: examples/stable_diffusion_v2/tools/data_check/data_check_part.py
Comment (VictorHe-1) [reply examples/stable_diffusion_v2/tools/data_check/data_check_part.py:1]: Moved

File: examples/stable_diffusion_v2/train_text_to_image.py (hunk 2 above)
Comment (VictorHe-1) [reply examples/stable_diffusion_v2/train_text_to_image.py:72]: format changed

**Review by zhanghuiyao — APPROVED** [review]

File: examples/stable_diffusion_v2/tools/data_utils/data_check_part.py [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+import multiprocessing as mp
+import os
+import tarfile
+
+import moxing as mox
+
+
+class RootFilter(logging.Filter):
+    def filter(self, record):
+        # 过滤掉根logger的WARNING、ERROR信息，避免mox.file.copy_parallel重复打印
```
Comment (SamitHuang) [inline examples/stable_diffusion_v2/tools/data_utils/data_check_part.py:25]: remove CN comments

**Review by SamitHuang — APPROVED** [review]

**Pull request merged**